crates/assembly-syntax/src/sema/mod.rs

```diff
@@ -179,9 +186,16 @@ fn visit_items(module: &mut Module, analyzer: &mut AnalysisContext) -> Result<()
                 }
 
                 // Evaluate all named immediates to their concrete values
+                log::debug!(target: "const-eval", "visiting procedure {}", procedure.name());
```

Comment:
Question (and not for this PR): do we want to keep logging/debugging statements in the code long-term?

Reply:
And do we not want to upgrade them to [tokio-tracing spans](https://github.com/tokio-rs/tracing), which the line @bobbinth is here commenting on is a posterchild use case for?

Reply:
These are specifically designed to be left in the code - they are zero-cost if logging isn't enabled (or the min log level is higher than the level of the log statement). Since they are particularly useful for debugging the inner workings of the assembler, it's worth keeping them around IMO.